photosynthesis_metrics/vif.py

```diff
@@ -89,17 +92,38 @@ def vif_p(prediction: torch.Tensor, target: torch.Tensor,
         sigma_trgt_pred = F.conv2d(target * prediction, kernel) - mu_trgt_pred
         
         # Zero small negative values
-        torch.relu_(sigma_trgt_sq)
-        torch.relu_(sigma_pred_sq)
-        
+        sigma_trgt_sq = torch.relu(sigma_trgt_sq)
+        sigma_pred_sq = torch.relu(sigma_pred_sq)
+
         g = sigma_trgt_pred / (sigma_trgt_sq + EPS)
         sigma_v_sq = sigma_pred_sq - g * sigma_trgt_pred
 
+        g = torch.where(sigma_trgt_sq >= EPS, g, torch.zeros_like(g))
+        sigma_v_sq = torch.where(sigma_trgt_sq >= EPS, sigma_v_sq, sigma_pred_sq)
+        sigma_trgt_sq = torch.where(sigma_trgt_sq >= EPS, sigma_trgt_sq, torch.zeros_like(sigma_trgt_sq))
+
+        g = torch.where(sigma_pred_sq >= EPS, g, torch.zeros_like(g))
+        sigma_v_sq = torch.where(sigma_pred_sq >= EPS, sigma_v_sq, torch.zeros_like(sigma_v_sq))
+
+        sigma_v_sq = torch.where(g >= 0, sigma_v_sq, sigma_pred_sq)
+        g = torch.relu(g)
+
+        sigma_v_sq = torch.where(sigma_v_sq > EPS, sigma_v_sq, torch.ones_like(sigma_v_sq) * EPS)
+    
         pred_vif_scale = torch.log10(1.0 + (g ** 2.) * sigma_trgt_sq / (sigma_v_sq + sigma_n_sq))
-        prediction_vif += torch.sum(pred_vif_scale, dim=[1, 2, 3])
-        target_vif += torch.sum(torch.log10(1.0 + sigma_trgt_sq / sigma_n_sq), dim=[1, 2, 3])
+        prediction_vif = prediction_vif + torch.sum(pred_vif_scale, dim=[1, 2, 3])
+        target_vif = target_vif + torch.sum(torch.log10(1.0 + sigma_trgt_sq / sigma_n_sq), dim=[1, 2, 3])
+
+    vif = (prediction_vif + EPS) / (target_vif + EPS)
 
-    return prediction_vif / target_vif
+    # Reduce if needed
+    if reduction == 'mean':
+        return vif.mean(dim=0)
+    elif reduction == 'sum':
+        return vif.sum(dim=0)
+    elif reduction != 'none':
+        raise ValueError(f'Expected reduction modes are "mean"|"sum"|"none", got {reduction}')
+    return vif
```

Comment:
Consider reimplmeneting it [the way it is done in BRISQUE](https://github.com/photosynthesis-team/photosynthesis.metrics/blob/83e65bb48e0e61b6b89ae75aaa3fcbb63b74dac5/photosynthesis_metrics/brisque.py#L178)